### DISCUSSION TITLE
Mocha scss issue

### DISCUSSION THREAD

**Discussion by nikos-manolas** [vercel/next.js#14583] [Help] [answered]
Hey,

I am using scss modules
Trying to configure mocha tests.. 
Using the example from https://github.com/vercel/next.js/tree/canary/examples/with-mocha.
Tests are complaining: `@import 'all';
^
SyntaxError: Invalid or unexpected token`
I assume I need to compile the scss somehow? 

Any help would be much appreciated.
Thank you.

**Comment by jamesmosier** [comment, marked as answer]
This [Stack Overflow](https://stackoverflow.com/a/34375878/1554860) answer worked for me:

`npm install --save-dev ignore-styles`

then run your tests in your `package.json` via:

`"test": "cross-env NODE_ENV=test mocha --require ignore-styles",`

**Reply by nikos-manolas** [reply]
Thank you for the answer.
That works if I don't want to import any styles. 
What if I want in my unit tests to shallow.find based on the classNames of the components? This wouldn't work as they will be undefined. Is there a way to use the imports?

**Reply by jamesmosier** [reply]
For that you'd want to add this to `mocha.setup.js` (found via [this gist](https://gist.github.com/mmrko/288d159a55adf1916f71))

```js
import hook from 'css-modules-require-hook'
import sass from 'node-sass'

hook({
  extensions: [ '.scss' ],
  preprocessCss: data => sass.renderSync({ data }).css
})
```

and also add `@babel/register` and `css-modules-require-hook` to your `package.json`

then you'd get something like this:

```jsx
<div>
  <p className="_index_module__foo">
    Hello World!
  </p>
</div>
```

**Reply by nikos-manolas** [reply]
Will try it out. Thank you.
